fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -341,14 +343,51 @@ private void pruneTables(OptExpression tree, TaskContext rootTaskContext, Column
             ruleRewriteIterative(tree, rootTaskContext, new MergeTwoProjectRule());
             rootTaskContext.setRequiredColumns(requiredColumns.clone());
             ruleRewriteOnlyOnce(tree, rootTaskContext, RuleSetType.PRUNE_COLUMNS);
-            context.setEnableLeftRightJoinEquivalenceDerive(true);
             ruleRewriteIterative(tree, rootTaskContext, RuleSetType.PUSH_DOWN_PREDICATE);
         }
     }
 
+    private void ruleBasedMaterializedViewRewriteMultiStages(OptExpression tree,
+                                                             TaskContext rootTaskContext,
+                                                             ColumnRefSet requiredColumns) {
+        if (!mvRewriteStrategy.mvStrategy.isMultiStages()) {
+            return;
+        }
+
+        OptimizerTraceUtil.logOptExpression("before RuleBasedMaterializedViewRewrite:\n%s", tree);
+
+        ///////////// Normalize query plan before rule based mv rewrite. /////////////
+        ruleRewriteOnlyOnce(tree, rootTaskContext, RuleSetType.PARTITION_PRUNE);
+        ruleRewriteIterative(tree, rootTaskContext, new MergeTwoProjectRule());
+        ruleRewriteIterative(tree, rootTaskContext, new MergeProjectWithChildRule());
+        CTEUtils.collectForceCteStatisticsOutsideMemo(tree, context);
+        rootTaskContext.setRequiredColumns(requiredColumns.clone());
+
+        // TODO: disable join reorder for both query and mv's plan for better match and optimizer cost.
+
+        // try view based mv rewrite first, then try normal mv rewrite rules
+        if (mvRewriteStrategy.enableRBOViewBasedRewrite) {
+            // view based mv rewrite for single view
+            viewBasedMvRuleRewrite(tree, rootTaskContext);
```

Comment:
seems duplicated with ruleBasedMaterializedViewRewrite
better to refactor it
better to not repeat youself.